lib/puppet/type/archive.rb

```diff
@@ -234,6 +234,15 @@ def should_to_s(value)
     defaultto :false
   end
 
+  newparam(:cacert_file) do
+    desc 'path to a custom CA certificate bundle file'
+    validate do |value|
+      unless !value.nil? && Puppet::Util.absolute_path?(value)
```

Comment:
Is this correct? I feel like it should be:
```suggestion
      if !value.nil? && !Puppet::Util.absolute_path?(value)
```

However, I'm having a hard time wrapping my brain on it. It would be great if you could just say `type 'Optional[Stdlib::AbsolutePath]'` in the type DSL.